modules/openxr/openxr_api.cpp

```diff
@@ -1175,8 +1179,8 @@ bool OpenXRAPI::obtain_swapchain_formats() {
 		}
 
 		if (color_swapchain_format == 0) {
-			color_swapchain_format = usable_swapchain_formats[0]; // just use the first one and hope for the best...
-			print_line("Couldn't find usable color swap chain format, using", get_swapchain_format_name(color_swapchain_format), "instead.");
+			color_swapchain_format = supported_swapchain_formats[0]; // just use the first one and hope for the best...
```

Comment:
Why does this switch from `usable_swapchain_format[0]` to `supported_swapchain_format[0]`? This would seem to mean we could select a swapchain format the OpenXR runtime supports, but which we don't think should be used for the color swapchain. I think with this we could end up trying to use a depth format for the color swapchain, or something like that?

Reply:
Thats actually a bug fix. If none of the swapchain formats we want is supported, we were chosing the first swapchain format we want (which we know is not supported), so it would fail to create the swapchain (because it's not supported).

So I changed it that if none of the swapchain formats we want is supported, we choose the first supported swapchain format, and hope for the best.

Honestly, there is a part of me that feels we should just fail in this situation.

Reply:
Ah, ok

> Honestly, there is a part of me that feels we should just fail in this situation.

Personally, I'd be in favor of that. If this code runs, unexpected things will happen, and not even the same unexpected thing over time, given that an OpenXR runtime can change the list of supported formats (or even just the order of the formats) at any time

Reply:
Ok, changed them to errors. Indeed makes far more sense.